### PR TITLE
Lazyload social embeds

### PR DESCRIPTION
**Pull request by chris-hinds**
Resolves N/A

Social embeds can be large, especially YouTube when it loads in hundreds of KB of JS for videos that may never been seen or is at least not required right away. When a YouTube embed loads its JS the JS is parsed right away even though the embedded player has not been seen/interacted with yet.

This PR prevents SocialEmbeds from loading until they are scrolled into view.

**Overall change:**
Uses the already installed react-lazyload package to Lazyload social embeds that are below the fold

**Code changes:**

- Wraps the canonical social embed component with react-lazyload

---

- [ ] I have assigned myself to this PR and the corresponding issues
- [ ] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [ ] I have assigned this PR to the Simorgh project

**Testing:**

- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`)
- [ ] This PR requires manual testing

### CODE REVIEW

File: src/app/containers/SocialEmbed/index.jsx [hunk 1]
```diff
@@ -82,14 +83,16 @@ const SocialEmbedContainer = ({ blocks }) => {
...
         ) : (
-          <CanonicalSocialEmbed
-            provider={provider}
-            service={service}
-            oEmbed={oEmbed}
-            fallback={fallback}
-            skipLink={skipLink}
-            caption={caption}
-          />
+          <Lazyload once>
```
Comment (karinathomasbbc) [inline src/app/containers/SocialEmbed/index.jsx]: Would we expect any tests to be updated as a result of this?

Comment (chris-hinds) [reply src/app/containers/SocialEmbed/index.jsx]: Snapshots will probably need updating, just letting CI run to find out

**Review by karinathomasbbc — APPROVED** [review]

**Comment by greenc05** [comment]
Can we check that screen readers can still interact ok please @chris-hinds?

**Comment by chris-hinds** [comment]
> Can we check that screen readers can still interact ok please @chris-hinds?

Will do.

**Comment by greenc05** [comment]
Dev a11y swarm requirements:

- Navigating through all elements from the top of a page to the social embed - Does the social embed come into view as expected with the screen reader and is announced
- Navigating via links menu from the top of the page - Is the social embed content listed if not in view? I assume not? Are we ok with this? Use case - A user comes to a story after perhaps visiting it before and looks for the social embed…. Though I assume that none AT users wouldn’t be able to find it either in this case e.g. via a ‘Find’ - Let’s test ‘Find’ too. Even if we are ok with these things, we should make a note of them and be aware…
- Navigating via the keyboard tab key… Does the social embed content come into view and part of the page tab order as expected? (the skip link is the first content of a social embed)

**Comment by chris-hinds** [comment]
I tested the the same asset on live and then with this feature branch using Voice Over for Mac and IOS. The functionality is the same across both. I am able to navigate the site and see/hear the skip link for the social embed as I navigated through the page.

**Comment by paruchurisilpa** [comment]
I tested using the asset http://simorgh-infrastructure.api.bbci.co.uk/mundo/noticias-internacional-54366422 on JAWS and talkback. The functionality is the same across both. I am able to navigate throug the site without issues and and  skip link works as expected..

**Comment by greenc05** [comment]
@chris-hinds @paruchurisilpa should I assume that the actual social embed was read out after the skip link as per usual?

Just NVDA screen reader left to test for point 1 above? https://github.com/bbc/simorgh/pull/7874#issuecomment-697597517

Would be interesting to know the outcome of points 2 and 3 from above when you have a moment please

**Comment by chris-hinds** [comment]
@greenc05 > should I assume that the actual social embed was read out after the skip link as per usual?

That is correct, none of this logic has changed.

will address the other points later.

**Comment by chris-hinds** [comment]
When navigating the page using the links menu in Voice Over, if the social embed has not been scrolled into view it's links will not appear in the links menu. This is expected as the as the embed container has yet to be rendered.

A future improvement would be to ensure we always render the embed container with the `skip link`. This way when clicking the 'skip link' in the links menu the user would be scrolled to the embed container and the embed would render.

**Comment by paruchurisilpa** [comment]
Checked using NVDA for the a11y requirements mentioned and using https://bbc.github.io/accessibility-news-and-you/accessibility-and-testing-with-nvda.

LGTM, didn't notice anything unusual..

**Review by chewu — APPROVED** [review]

**Pull request merged**